src/steps/command.ts

```diff
@@ -145,13 +164,15 @@ export class CommandStep extends LabeledStep {
         }
     }
 
-    add(c: string | Plugin | Command): this {
+    add(c: string | Plugin | Command | MetaCommand): this {
         if (c instanceof Plugin) {
             this.plugins.add(c);
         } else if (typeof c === 'string') {
             this.command.push(new Command(c));
         } else if (c instanceof Command) {
```

Comment:
this should already accept the above, assuming that each `MetaCommand` is a subclass of a `Command` (which I think is something we'd want, otherwise you could pass a completely unconnected `MetaCommand` in here)

Reply:
I'm not sure if we can do this, since `Command` doesn't support chaining but `MetaCommand` does, so their constructors are different

Reply:
as long as the `Command` constructor is satisfied with a `super` call it should be okay. It is possible that we need to replace some ocurrences of `c: Command` with `c: ? extends Command` - we can do that, how about we get this PR in, but with a sample command that is not part of core?

Reply:
> as long as the Command constructor is satisfied with a super call it should be okay.

yeah, I don't think it is since the `Command` constructor can only take in a string, and `MetaCommand` takes in `string | Command`.

> how about we get this PR in, but with a sample command that is not part of core?

Do you mean remove the `RetryCommand` and define it in examples?

Reply:
> yeah, I don't think it is since the `Command` constructor can only take in a string, and `MetaCommand` takes in `string | Command`.

well each `Command` has a `.serialize` that returns a string after #65 - but it any case, we can widen what the `Command` constructor takes if need be :)

> Do you mean remove the RetryCommand and define it in examples?

yeah, use what you introduced in #65 and show how to use a custom meta command that can be used with `buildkite-graph` without making it part of the core.

Reply:
> well each Command has a .serialize that returns a string after #65 - but it any case, we can widen what the Command constructor takes if need be :)

I initially tried doing it this way, but `Command` doesn't support chaining, and I don't think it makes sense to change it so it does or we'd have something like new Command(new Command("yarn")). If `Command` doesn't support chaining then the constructor can only be Command(command: string), which means the super call in MetaCommand will fail.

> yeah, use what you introduced in #65 and show how to use a custom meta command that can be used with buildkite-graph without making it part of the core.

👍

Reply:
> I initially tried doing it this way, but `Command` doesn't support chaining, and I don't think it makes sense to change it so it does or we'd have something like new Command(new Command("yarn")). If `Command` doesn't support chaining then the constructor can only be Command(command: string), which means the super call in MetaCommand will fail.

I don't think that is true:

```ts
class Command {
    constructor(private readonly c: string) {

    }

    toString() {
        return this.c.toString();
    }
}

abstract class MetaCommand extends Command {
    constructor(c: Command) {
        super(c.toString())
    }
}


class RetryCommand extends MetaCommand {
    constructor(retries: number, c: Command) {
        super(new Command(new Array(retries).fill(c.toString()).join(' || ')));
    }
}


class Step {
    constructor(private readonly c: Command) {

    }

    toString() {
        return this.c.toString();
    }
}


const command = new RetryCommand(2, new Command('yarn'));
const step = new Step(command);
console.log(step.toString()); // prints "yarn || yarn"
```
see: https://www.typescriptlang.org/play?#code/MYGwhgzhAEDCD2BbRYB2ATaBvAUNf0w8qEALgE4Cuwp85AFAA7kCWAbmKQKbTldjpiIAJ6EAXNDKtUAcwCU2HHgIBfJQWi0AyhRaz6C3Bo19Slcqk0ALFhAB0wO9t365AbmX41anGABGUmA0hOBQ0ACyXKRgCMhomFwAHtwYMLEoGIoaRCQU1LQMwBLp8YaexhCUjFyFTvA60jIGcuXeSjigkDAASlHkwiWZSSnoMJHRg5hGBDlS+XT0pqxcEBKolIh+NQA04nBIGehlxgSV1QyoXADu+3EY9Jc3AILk5GDCi30sK3J2AGYsEAgeiOZyNZp2ABW8D09AA5NAAD6I6BwuTojwaNpKTphHRcRhZGbEOY0BbMdicHh8ARCURFW6HMqtdQEMF6JrHE68KLmSykGz2UH1FyczGqHA+Dok0iEA7xaAAXmgj2gvQoA3l9wATLtVZN4cIwBY0e5pblJNxCcrVfjGCCtUcPLN4CAuHYQPAmmQCXUGhzmm4gA (simplified implementation of a retry command)

Reply:
Ah adding `private readonly` and removing `public` in `MetaCommand` makes it work, thanks!